server/procedure_run_collection.cc

```diff
@@ -44,10 +45,126 @@ initRoutes(RouteManager & manager)
 
     ExcAssert(manager.getKey);
 
-    manager.addPutRoute();
```

Comment:
The boilerplace was copied and modified to handle the update of a procedure run.  This is clearly not ideal.  I will try to minimize the code duplication.